driver/mounter.go

```diff
@@ -208,6 +232,26 @@ func (m *mounter) Unmount(target string) error {
 	return mount.CleanupMountPoint(target, m.kMounter, true)
 }
 
+func (m *mounter) IsAttached(source string) (bool, error) {
+	out, err := m.attachmentValidator.evalSymlinks(source)
+	if err != nil {
+		return false, fmt.Errorf("error evaluating the symbolic link %q: %s", source, err)
+	}
+
+	_, deviceName := filepath.Split(out)
+	deviceStateFilePath := fmt.Sprintf("/sys/class/block/%s/device/state", deviceName)
+	deviceStateFileContent, err := m.attachmentValidator.readFile(deviceStateFilePath)
+	if err != nil {
+		return false, fmt.Errorf("error reading the device state file %q: %s", deviceStateFilePath, err)
+	}
+
+	if string(deviceStateFileContent) != fmt.Sprintf("%s\n", runningState) {
```

Comment:
It'd be nice if we returned the content of the file when it does not match to ease troubleshooting. Returning an error for this case as well would achieve that goal (which would then also render the boolean return type obsolete).

Alternatively, we can pass in a logger for this scenario.

I am slightly preferring the error approach since it seems a bit easier to use and test against, but no strong opinion. Will leave the final decision to you.

Reply:
I'll return an error with the following message to ease troubleshooting. 

```go 
return false, fmt.Errorf("error comparing the state file content, expected: %s, got: %s", runningState, string(deviceStateFileContent))
```